### ISSUE TITLE
`token_endpoint_auth_method` incorrectly set in client example (v1 beta)

### ISSUE THREAD

**Issue by rexwangcc** [auth0/terraform-provider-auth0#813]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

While https://registry.terraform.io/providers/auth0/auth0/latest/docs/resources/client still has `token_endpoint_auth_method` on the example struct, when we tf plan the resource, it cries about:

```
│ Error: Unsupported argument
│ 
│   on auth0.tf line 4, in resource "auth0_client" "xxx":
│    4:   token_endpoint_auth_method = "client_secret_post"
│ 
│ An argument named "token_endpoint_auth_method" is not expected here.
```

### Expectation

The doc should be updated?

### Reproduction

Write a resource like:

```hcl
resource "auth0_client" "xxxx" {
  name                       = "XXX"
  app_type                   = "regular_web"
  token_endpoint_auth_method = "client_secret_post"
  callbacks                  = ["http://localhost:3000/oidc/login-callback", "https://app-example.com/oidc/login-callback"]
  allowed_logout_urls        = ["http://localhost:3000/landing", "https://example.com"]
  oidc_conformant            = true
  grant_types                = ["authorization_code", "refresh_token"]
  jwt_configuration {
    lifetime_in_seconds = 300
    alg                 = "RS256"
  }
...
```

and try to plan the resource

### Auth0 Terraform Provider version

1.0.0-beta.3

### Terraform version

~> 1.5.0

**Comment by fabrizio-akoya** [comment]
I'm running into this as well. We're in the process of migrating our rules to actions in anticipation of rules being decommissioned in November 2024. As part of this migration, we've discovered we have to use node runtime 18 instead of 16, which we had been using. And we've learned that we can't upgrade the runtime to v18 via terraform unless we upgrade the Auth0 tf provider from 0.42 (which we were currently using), to a 1.0.0-beta version. So now that I've upgraded the terraform provider, we're running into errors generating the plan, including the error mentioned here.

**Comment by willvedd** [comment]
@rexwangcc You are correct, the `token_endpoint_auth_method` field in the example is erroneously listed. It was removed from the client resource in v1.0.0-beta and now managed through the `auth0_client_credentials` resource. This is mentioned in the [v0=>v1 migration guide](https://github.com/auth0/terraform-provider-auth0/blob/v1/MIGRATION_GUIDE.md#client-authentication-method). I have made this update in #822.

> As part of this migration, we've discovered we have to use node runtime 18 instead of 16, which we had been using. And we've learned that we can't upgrade the runtime to v18 via terraform unless we upgrade the Auth0 tf provider from 0.42 (which we were currently using), to a 1.0.0-beta version.
@fabrizio-akoya Heads up – we've released [v0.50.2](https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.50.2) which allows users to use node 18 runtime in actions without needing to upgrade to v1. However, we strongly suggest you to upgrade to v1 as soon as convenient because we do not plan to make further updates to v0.

Thanks for the feedback and hope this clears things up for both of you!

**Comment by rexwangcc** [comment]
> @rexwangcc You are correct, the `token_endpoint_auth_method` field in the example is erroneously listed. It was removed from the client resource in v1.0.0-beta and now managed through the `auth0_client_credentials` resource. This is mentioned in the [v0=>v1 migration guide](https://github.com/auth0/terraform-provider-auth0/blob/v1/MIGRATION_GUIDE.md?rgh-link-date=2023-09-08T20%3A47%3A25Z#client-authentication-method). I have made this update in #822.
> 
> > As part of this migration, we've discovered we have to use node runtime 18 instead of 16, which we had been using. And we've learned that we can't upgrade the runtime to v18 via terraform unless we upgrade the Auth0 tf provider from 0.42 (which we were currently using), to a 1.0.0-beta version.
> 
> @fabrizio-akoya Heads up – we've released [v0.50.2](https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.50.2) which allows users to use node 18 runtime in actions without needing to upgrade to v1. However, we strongly suggest you to upgrade to v1 as soon as convenient because we do not plan to make further updates to v0.
> 
> Thanks for the feedback and hope this clears things up for both of you!

Thanks for the prompt reply!

**Comment by fabrizio-akoya** [comment]
Thanks for the clarifications @willvedd!
